OverlayPlugin.Updater/HttpClientWrapper.cs

```diff
@@ -50,28 +60,39 @@ public static string Get(string url)
                     else
                     {
                         var buffer = new byte[40 * 1024];
-                        var length = 0;
+                        long length = 0;
 
                         IEnumerable<string> lengthValues;
                         if (response.Headers.TryGetValues("Content-Length", out lengthValues))
                         {
-                            int.TryParse(lengthValues.GetEnumerator().Current, out length);
+                            long.TryParse(lengthValues.GetEnumerator().Current, out length);
+                        }
+                        else
+                        {
+                            long? @long = response.Content.Headers.ContentLength;
+                            if (@long.HasValue)
+                            {
+                                length = @long.Value;
+                            }
                         }
```

Comment:
Moving this to a new review comment since the new code is closer to correct.

Consider the following three possibilities:

1. No `Content-Length` header
 a. This hits the `else` branch, which does nothing because the header does not exist
2. One `Content-Length` header
 a. This hits the `if` branch, which works as before this PR
3. More than one `Content-Length` header
 a. This may or may not result in an exception and there's nothing we can do to prevent it

Given that we can't really prevent an error of some sort with multiple `Content-Length` headers, I think that this code makes the most sense after testing via the test code I posted on the previous comment:

```cs
                long length = 0;

                long? nLength = response.Content.Headers.ContentLength;

                if (nLength.HasValue)
                {
                    length = nLength.Value;
                }
```